packages/mui-base/src/utils/useHostElementName.ts

```diff
@@ -0,0 +1,50 @@
+'use client';
+import * as React from 'react';
+
+type UseHostElementNameParameters = {
+  /**
+   * The HTML element expected to be rendered, e.g. 'div', 'button' etc
+   * @default ''
+   */
+  hostElementName?: keyof HTMLElementTagNameMap;
+  /**
+   * The name of the component using useControllableReducer.
+   * For debugging purposes.
+   */
+  componentName?: string;
+};
+/**
+ *
+ * API:
+ *
+ * - [useHostElementName API](https://mui.com/base-ui/api/use-host-element-name/)
```

Comment:
Do we want to make it public? I'd add the @ignore to jsdocs, so it doesn't appear in the docs.